File: BedwyrKnx.GUI/Forms/frmAuthModify.vb
Imports Ouroboros.AuthManager.Eos

Public Class frmAuthModify

    Private Sub btnOK_Click(sender As Object, e As EventArgs) Handles btnOK.Click
        If String.IsNullOrWhiteSpace(tbInput.Text) Then Exit Sub
        If chkModeNew.Checked Then
            Try
                Dim auth As New AuthInfo(tbInput.Text)
                If auth.CreateFile() Then
                    MessageBox.Show("授权成功，请重启程序", "Info", MessageBoxButtons.OK, MessageBoxIcon.Information)
                End If
            Catch ex As Exception
                MessageBox.Show(ex.Message, "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning)
            End Try
        ElseIf chkModUpdate.Checked Then
            Try
                If _AuthInfo.Update(tbInput.Text) Then
                    MessageBox.Show("授权成功，请重启程序", "Info", MessageBoxButtons.OK, MessageBoxIcon.Information)
                End If
            Catch ex As Exception
                MessageBox.Show(ex.Message, "Warning", MessageBoxButtons.OK, MessageBoxIcon.Warning)
            End Try
        Else
            Environment.Exit(0)
        End If
        Me.DialogResult = DialogResult.OK
        Me.Close()
    End Sub

    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        Me.DialogResult = DialogResult.Cancel
        Me.Close()
    End Sub

    Private Sub frmAuthModify_Closed(sender As Object, e As EventArgs) Handles Me.Closed
        Environment.Exit(0)
    End Sub

End Class

File: BedivereKnx.GUI/Forms/FrmAuthModify.cs
using Ouroboros.AuthManager.Eos;

namespace BedivereKnx.GUI.Forms
{

    public partial class FrmAuthModify : Form
    {

        public FrmAuthModify()
        {
            InitializeComponent();
        }

        private void btnOK_Click(object sender, EventArgs e)
        {
            if (string.IsNullOrWhiteSpace(tbInput.Text)) return;
            string input = tbInput.Text.Trim();
            if (chkModeNew.Checked) //全新授权
            {
                try
                {
                    AuthInfo authInfo = new(input);
                    if (authInfo.CreateFile())
                    {
                        MessageBox.Show(Resources.Strings.Msg_AuthSuccess, "Info", MessageBoxButtons.OK, MessageBoxIcon.Information);
                    }
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            else if (chkModUpdate.Checked) //授权升级
            {
                try
                {
                    if (Globals.AuthInfo!.Update(input))
                    {
                        MessageBox.Show(Resources.Strings.Msg_AuthSuccess, "Info", MessageBoxButtons.OK, MessageBoxIcon.Information);
                    }
                }
                catch (Exception ex)
                {
                    MessageBox.Show(ex.Message, "Error", MessageBoxButtons.OK, MessageBoxIcon.Error);
                }
            }
            else //非法绕过模式选择的情况
            {
                Environment.Exit(-4);
            }
            DialogResult = DialogResult.OK;
            Close();
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void FrmAuthModify_FormClosed(object sender, FormClosedEventArgs e)
        {
            Environment.Exit(-4);
        }

    }

}
